components/uniset/src/uniset.rs

```diff
@@ -0,0 +1,328 @@
+use std::{boxed::Box, error::Error, iter::Iterator, slice::Iter, vec::Vec};
+
+const CODEPOINT_MAX: u32 = 0x10FFFF; // does max imply inclusive? else should be 10FFFF
+const CODEPOINT_MIN: u32 = 0x000000;
+const BMP_MAX: u32 = 0xFFFF;
+
+/// Given string representation of inversion list create set
+///
+/// See UnicodeSet::new for conditions
+fn parse_serial_string(serialize_str: &str) -> Result<Vec<u32>, Box<dyn Error>> {
+    let mut serialize = serialize_str.split(" ");
+    let capacity: usize = serialize
+        .next()
+        .ok_or("Splitting did not yield anything")?
+        .parse()?;
+    if capacity % 2 != 0 {
+        return Err("Capacity must be even".into());
+    }
+    let mut serialized_vec: Vec<u32> = Vec::with_capacity(capacity);
+    let mut prev: Option<u32> = None;
+    for str_elem in serialize {
+        let parsed: u32 = str_elem.parse()?;
+        if serialized_vec.len() + 1 > serialized_vec.capacity() {
```

Comment:
You are correct that vectors auto resize when growing and that `len()` could never be larger than `capacity()`  (not when removing elements, but that is not related to the line, just for future reference).

The check is to make sure that the vector does not grow in size from the user designated size.

My concern here is that if the user designates a vector size, when the environment has tight resource constraints, that is too big for their environment, the user will at least be notified in the above line: `let mut serialized_vec: Vec<u32> = Vec::with_capacity(capacity);`, otherwise we could have just initialized with `Vec::new()`. For this reason I wanted to make sure that the capacity of the vector remains as the user designated to avoid confusion.

Let me know if this sounds reasonable.